crypto/src/user_identity.rs

```diff
@@ -35,7 +35,7 @@ pub fn make_user_name_identity_token(channel_security_policy: SecurityPolicy, us
 
     // Table 179 Opc Part 4 provides a table of which encryption algorithm to use
     let security_policy = if channel_security_policy == SecurityPolicy::None {
-        if user_token_policy.security_policy_uri.is_empty() || token_security_policy != SecurityPolicy::None {
+        if user_token_policy.security_policy_uri.is_empty() || token_security_policy == SecurityPolicy::None {
```

Comment:
I think this entire if-statement can use some simplification. Some of the later clauses are a bit weird. But this is fine.